pairing_api/src/pairing.rs

```diff
@@ -0,0 +1,470 @@
+            .await?;
+
+        Ok(())
+    }
+
+    /// Private function to publish a request response.
+    pub async fn publish_response(
+        &self,
+        topic: &str,
+        params: ResponseParamsSuccess,
+        message_id: MessageId,
+        client: &Client,
+    ) -> Result<(), PairingClientError> {
+        let irn_metadata = params.irn_metadata();
+        let response = Response::Success(SuccessfulResponse {
+            id: message_id,
+            jsonrpc: JSON_RPC_VERSION_STR.into(),
+            result: serde_json::to_value(params)
+                .map_err(|err| PairingClientError::EncodeError(err.to_string()))?,
+        });
+
+        // Publish the encrypted message
+        self.publish_payload(topic, irn_metadata, Payload::Response(response), client)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Private function to publish a payload.
+    async fn publish_payload(
+        &self,
+        topic: &str,
+        irn_metadata: IrnMetadata,
+        payload: Payload,
+        client: &Client,
+    ) -> Result<(), PairingClientError> {
+        // try to extend session before updating local store.
+        let sym_key = {
+            let pairings = self.pairings.lock().await;
+            let pairing = pairings
+                .get(topic)
+                .ok_or_else(|| PairingClientError::PairingNotFound)?;
+            pairing.sym_key
+        };
+
+        let payload = serde_json::to_string(&payload)
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+        let message = encrypt_and_encode(EnvelopeType::Type0, payload, &sym_key)
+            .map_err(|err| PairingClientError::EncodeError(err.to_string()))?;
+
+        // Publish the encrypted message
+        {
+            client
+                .publish(
+                    topic.into(),
+                    message,
+                    None,
+                    irn_metadata.tag,
+                    Duration::from_secs(irn_metadata.ttl),
+                    irn_metadata.prompt,
+                )
+                .await?;
+        };
+
+        Ok(())
+    }
+
+    /// Private function to generate a WalletConnect URI.
+    fn generate_uri(pairing: &PairingInfo, sym_key: &str) -> String {
+        let mut url = format!(
+            "wc:{}@{}?symKey={}&relay-protocol={}&expiryTimestamp={}",
+            pairing.topic, VERSION, sym_key, pairing.relay.protocol, pairing.expiry
+        );
+
+        if !pairing.methods.0.is_empty() {
+            let methods_str = pairing
+                .methods
+                .0
+                .iter()
+                .map(|method_group| format!("[{}]", method_group.join(",")))
+                .collect::<Vec<_>>()
+                .join(",");
+
+            url.push_str(&format!("&methods={methods_str}"));
+        };
+
+        url
+    }
+}
+
+fn gen_sym_key() -> [u8; 32] {
+    OsRng.gen::<[u8; 32]>()
+}
```

Comment:
If we can use `try_fill_bytes` https://github.com/KomodoPlatform/komodo-defi-framework/blob/f487947c9d8f454969396986f8f259961ce87f6b/mm2src/common/common.rs#L837 it will be a bit safer.

Reply:
done. Thanks